parse.go

```diff
@@ -63,19 +63,23 @@ func NewParserFromBytes(data []byte, frameChannel chan *frame.Frame) (Parser, er
 // NewParserFromFile initializes and returns a new dicom Parser from a file path
 func NewParserFromFile(path string, frameChannel chan *frame.Frame) (Parser, error) {
 	file, err := os.Open(path)
+	defer file.Close()
 	if err != nil {
 		return nil, err
 	}
+
 	st, err := file.Stat()
 	if err != nil {
 		return nil, err
 	}
-	p, err := NewParser(file, st.Size(), frameChannel)
+
+	data := make([]byte, st.Size())
+	_, err = file.Read(data)
```

Comment:
The main issue with this approach is that we must read the entire file into memory at once. By using a buffered `io.Reader` as before, we can instead efficiently read the file as we go, particularly useful if you're reading data off of a network filesystem. Because of the frame channel, this library supports end to end streaming--that is, we can be emitting parsed image frames before we have even read the entire DICOM from the source into memory. That would not be possible with this change.

What I think we should do instead is keep NewParserFromFile as is, but close the file explicitly if we return with other errors (e.g. in the file.Stat, or when creating the parser). 

The other option is to deprecate NewParserFromFile entirely, and rely on the user to provide you an io.Reader and size directly, which is what I'm contemplating in the 1.0 rewrite API (see https://github.com/suyashkumar/dicom/blob/s/1.0-rewrite/parse.go#L36)  

This way, it's up to the caller to manage the file itself.

Reply:
@suyashkumar I reverted back to the first commit I made. This should entail what you discussed in your comment. When it encounters an error, it explicitly closes the file to avoid leaving them open in memory, but also doesn't alter the way that NewFileParser is created. Let me know if I missed anything or if there are any other changes I can make.

Thank you for working with me on this!

Reply:
Thanks, LGTM!